packages/netlify-cms-backend-azure/src/API.ts

```diff
@@ -0,0 +1,782 @@
+    const mergeRequests = await this.getMergeRequests(branch);
+    if (mergeRequests.length <= 0) {
+      throw new EditorialWorkflowError('content is not under editorial workflow', true);
+    }
+
+    return mergeRequests[0];
+  }
+
+  async getDifferences(to: string) {
+    const result = await this.requestJSON({
+      url: `${this.endpointUrl}/diffs/commits`,
+      params: {
+        baseVersion: this.branch,
+        targetVersion: this.refToBranch(to),
+      },
+    });
+
+    return result.changes;
+  }
+
+  async editorialWorkflowGit(files: (Entry | AssetProxy)[], entry: Entry, options: PersistOptions) {
+    const contentKey = this.generateContentKey(options.collectionName as string, entry.slug);
+    const branch = this.branchFromContentKey(contentKey);
+    const unpublished = options.unpublished || false;
+
+    if (!unpublished) {
+      const items = await this.getCommitItems(files, this.branch);
+
+      await this.uploadAndCommit(items, options.commitMessage, branch, true);
+
+      await this.createPullRequest(
+        branch,
+        options.commitMessage,
+        options.status || this.initialWorkflowStatus,
+      );
+    } else {
+      const items = await this.getCommitItems(files, branch);
+      await this.uploadAndCommit(items, options.commitMessage, branch, true);
+    }
+  }
+
+  /**
+   * Gets a pull request and updates labels to allow it to move between different
+   * states in the editorial workflow.
+   * @param collection The Jekyll collection the item is in.
+   * @param slug The slug for the content item.
+   * @param newStatus The new status for the item.
+   */
+  async updateUnpublishedEntryStatus(collection: string, slug: string, newStatus: string) {
+    const contentKey = this.generateContentKey(collection, slug);
+    const branch = this.branchFromContentKey(contentKey);
+
+    const mergeRequest = await this.getBranchMergeRequest(branch);
+
+    const labels = [
+      ...mergeRequest.labels
+        .filter((label: AzurePRLabel) => !isCMSLabel(label.name))
+        .map((label: AzurePRLabel) => label.name),
+      statusToLabel(newStatus),
+    ];
+
+    await this.updatePullRequestLabels(mergeRequest, labels);
+  }
+
+  async deleteUnpublishedEntry(collectionName: any, slug: string) {
+    const contentKey = this.generateContentKey(collectionName, slug);
+    const branch = this.branchFromContentKey(contentKey);
+    const mergeRequest = await this.getBranchMergeRequest(branch);
+    await this.abandonPullRequest(mergeRequest);
+  }
+
+  async publishUnpublishedEntry(collectionName: string, slug: string) {
+    const contentKey = this.generateContentKey(collectionName, slug);
+    const branch = this.branchFromContentKey(contentKey);
+    const mergeRequest = await this.getBranchMergeRequest(branch);
+    await this.completePullRequest(mergeRequest);
+  }
+
+  async updatePullRequestLabels(mergeRequest: any, labels: string[]) {
+    mergeRequest.labels.forEach(async (l: AzurePRLabel) => {
```

Comment:
Using `async` inside `forEach` doesn't behave the way [you expect it to be](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop).

Also the next `forEach` seems redundant as we can just `POST` the CMS label.

A simpler way to implement this (since it seems the azure API doesn't support updating all labels in one request) is to change `updatePullRequestLabels` to `replaceLabel` and pass in `toReplace` and `newLabel`, delete the old one and posting the new one (or something similar).